### PR TITLE
Page/PluginPage: Add extensibility / customization props 

### PR DESCRIPTION
**Pull request by torkelo**
I do think we need more customization options for the header. At least give some pages room to explore alternative header designs (like a page header that you can click to edit an item name). Explorations we might later add to the core component as options. 

But I have been struggling a bit with how to add these, given where these elements are rendered and how it's mostly driven by props on NavModelItem. 

* Props on Page and PluginPage vs NavModelItem
* How to have a consistent way to override these for both main pages and item pages (PluginPage can only customize item page via pageNav, main page set via info in plugin.json right now, except subTitle) 
* Backward compatibility with how these customizations should work when topnav is disabled. Both these issues have lead me to a bit stuck in how to solve this. 

Maybe someone else can give it a go :)

### CODE REVIEW

**Comment by mckn** [comment]
Great initiative @torkelo! I will try some time to have a look at this in the beginning of next week, if that is ok?

**Comment by torkelo** [comment]
@mckn no need right now. @ashharrison90 is taking over the core part (making core Page extensible), think it then should be just adding the same props to PluginPage (if our current plan/props design works out)

**Comment by ashharrison90** [comment]
@torkelo pushed an update using the new props in the `PluginDetails` page.

this only applies with `topnav` on and it's still fairly rough, but if you wanna have a quick play around and let me know your thoughts that'd be great.

i'll work on splitting things out into separate PRs tomorrow 👍

File: public/app/features/plugins/admin/pages/PluginDetails.tsx [hunk 1]
```diff
@@ -27,9 +28,24 @@ export default function PluginDetails({ match, queryParams }: Props): JSX.Elemen
...
   const styles = useStyles2(getStyles);
+  const subtitle = (
+    <div className={styles.subtitle}>
+      <div>{plugin?.description}</div>
+      {plugin?.details?.links && plugin.details.links.length > 0 && (
+        <div className={styles.links}>
+          {plugin.details.links.map((link) => (
+            <a key={link.name} href={link.url} className="external-link">
+              {link.name}
+            </a>
```
Comment (torkelo) [inline public/app/features/plugins/admin/pages/PluginDetails.tsx:43]: could need a separator `|` here to make each link feel more destinct, a it currently look like one link "Website Licence" 
image

**Comment by torkelo** [comment]
@ashharrison90 this looks great! 

Now we just need to add the page info & actions bits to the old Page design (because they are now missing on the PluginDetails page when topnav is disabled) 

I would also add a align-items center on the actions to have them centered when there are pageInfo items, (does not have any effect when there are no page info items)

**Comment by ashharrison90** [comment]
closing in favour of https://github.com/grafana/grafana/pull/58509

**Pull request closed without merging**